src/pages/ViewEmployee/ViewEmployee.tsx

```diff
@@ -1,49 +1,98 @@
 import { useParams } from "react-router-dom";
 import SelectedSkills from "../../components/SelectedSkills/SelectedSkills";
 import DivWrapper from "./styles";
-import { filterArray } from "../../utils/filterArray";
-import { useEmployeeContext } from "../../contexts/EmployeeContext";
+import { useState, useEffect } from "react";
+import { getData } from "../../core/api";
+import { IEmployee } from "../../components/Table/types";
+import { CircularProgress } from "@mui/material";
+import { toast } from "react-toastify";
 
 const ViewEmployee = () => {
   const { employeeId } = useParams();
-  const {employeesData} = useEmployeeContext();
-  const [employee] = filterArray(employeesData, { id: [Number(employeeId)] });
+  const [loading, setLoading] = useState(true);
+  const [employee, setEmployee] = useState<IEmployee>({ skills: [] });
+  const fetchCurrentEmployeeData = async () => {
+    setLoading(true);
+    try {
+      const response = (await getData(`employee/${employeeId}`)).data.data;
+      setEmployee({
+        ...response,
+        department: response.department
+          ? response.department.department
+          : "N/A",
+        role: response.role ? response.role.role : "N/A",
+      });
+    } catch (error) {
+      toast.error(
+        `Employee details could not be fetched from server.`,
+        {
+          position: "top-center",
+          autoClose: 5000,
+          hideProgressBar: false,
+          closeOnClick: true,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
+          theme: "colored",
+        }
+      );
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (employeeId) fetchCurrentEmployeeData();
+  }, []);
+
   return (
     <DivWrapper className="page-content">
-      <div>
-        <h2>Employee Details</h2>
-      </div>
-      <p>
-        Name: <span>{employee.firstName + " " + employee.lastName}</span>
-      </p>
-      <p>
-        Date of Birth: <span>{employee.dob}</span>
-      </p>
-      <p>
-        Address: <span>{employee.address}</span>
-      </p>
-      <p>
-        Phone: <span>{employee.phone}</span>
-      </p>
-      <p>
-        Email: <span>{employee.email}</span>
-      </p>
-      <p>
-        Date of joining: <span>{employee.doj}</span>
-      </p>
-      <p>
-        Department: <span>{employee.department}</span>
-      </p>
-      <p>
-        Role: <span>{employee.role}</span>
-      </p>
-      <div>
-        <SelectedSkills
-          description="Skill(s)"
-          isView={true}
-          selectedSkills={employee.skills}
-        ></SelectedSkills>
-      </div>
+      {loading ? (
+        <div className="loader">
+          <CircularProgress />
+        </div>
+      ) : (
+        <>
+          <div>
+            <h2>Employee Details</h2>
+          </div>
+          <p>
+            Employee ID: <span>{employee.id}</span>
+          </p>
+          <p>
+            Name:{" "}
```

Comment:
do we really need this{""}

Reply:
I wanted a space to be added after the : symbol. Prettier extension by itself formatted the blank space into this form {" "}.